js/binance.js

```diff
@@ -6150,33 +6247,30 @@ module.exports = class binance extends Exchange {
         await this.loadMarkets ();
         let market = undefined;
         let method = undefined;
-        let defaultType = 'future';
         const request = {
             'incomeType': 'FUNDING_FEE', // "TRANSFER"，"WELCOME_BONUS", "REALIZED_PNL"，"FUNDING_FEE", "COMMISSION" and "INSURANCE_CLEAR"
         };
         if (symbol !== undefined) {
             market = this.market (symbol);
             request['symbol'] = market['id'];
-            if (market['linear']) {
-                defaultType = 'future';
-            } else if (market['inverse']) {
-                defaultType = 'delivery';
-            } else {
-                throw new NotSupported (this.id + ' fetchFundingHistory() supports linear and inverse contracts only');
+            if (!market['swap']) {
+                throw new NotSupported (this.id + ' fetchFundingHistory() supports swap contracts only');
             }
         }
+        let subType = undefined;
```

Comment:
if you look above few lines, there is `if (symbol !== undefined)`.
in that case, `market` object would already contain `subType` (`linear`, `inverse`) properties. So, maybe here, all these lines down below (the lines which try to define `type` and `subType` again) might be just inside `else` clause ?  otheriwse, if symbol is set by user, why down below we are re-determining `type` and `subtype` ?